second-commit

Senkid12 opened the pull request:
Четвертый спринт курса "Автоматизатор тестирования на java".
В тесте для создания заказа был использован WebDriverManager для FireFox

senkid committed:
second-commit

pavel-ulan commented on pom.xml:
```diff
@@ -0,0 +1,53 @@
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <source>8</source>
+          <target>8</target>
```
**надо исправить**
а почему отсылка к компилятору 8й версии? По условиям задания должна быть 11-я =)

pavel-ulan commented on src/main/java/org/yandex/prakrikum/pageObject/HomePage.java:
```diff
@@ -0,0 +1,61 @@
+    //    Элемент с верхней кнопкой "Заказать"
+    private final By upOrderButton = By.xpath("//*[@id=\"root\"]/div/div[1]/div[1]/div[2]/button[1]");
+    //    Элемент с нижней кнопкой "Заказать"
+    private final By downOrderButton = By.xpath("//*[@id=\"root\"]/div/div[1]/div[4]/div[2]/div[5]/button");
```
**надо исправить**
очень очень некрасивый локатор, состоящий из абсолютного пути =/

Старайся не использовать локаторы состоящие только из пути (не важно относительный/абсолютный). Как правило всегда можно подобрать другой, более читаемый, локатор. Такие локаторы плохо читаются, по ним непонятно, куда они указывают и если что-то меняется в верстке, то они ломаются и могут или никуда не вести или, что еще хуже, возвращать совершенно другой элемент.

pavel-ulan commented on src/main/java/org/yandex/prakrikum/pageObject/HomePage.java:
```diff
@@ -0,0 +1,61 @@
+    }
+    //Метод ожидания загрузки страницы
+    public void waitLoadHomePage() {
+        new WebDriverWait(driver, Duration.ofSeconds(10)).until(driver -> (driver.findElement(homePageHeader).getText() != null && !driver.findElement(homePageHeader).getText().isEmpty()));
```
мне просто интересно, а какую разницу ты ожидаешь от 
driver.findElement(homePageHeader).getText() != null 
и
!driver.findElement(homePageHeader).getText().isEmpty() ?

pavel-ulan commented on src/main/java/org/yandex/prakrikum/pageObject/PageAboutCustomer.java:
```diff
@@ -0,0 +1,50 @@
+    // Поле с телефоном заказчика
+    private final By fieldPhoneNumber = By.xpath("//*[@id=\"root\"]/div/div[2]/div[2]/div[5]/input");
+    //    Кнопка "Далее"
+    private final By nextButton = By.xpath("//*[@id=\"root\"]/div/div[2]/div[3]/button");
```
**надо исправить**
локаторы надо править

pavel-ulan commented on src/main/java/org/yandex/prakrikum/pageObject/constants/ScooterColors.java:
**хорошо**
клево, что использовал enumы и класс констант. И в целом здорово, что как-то систематизируешь используемые данные. Это хорошее решение, старайся придерживаться этого подхода.

pavel-ulan commented on src/test/java/org/yandex/prakrikum/pageObject/CreateOrderTest.java:
```diff
@@ -0,0 +1,92 @@
+        objPageAboutScooter.inputRentalForm(date, rentalPeriod, color, comment);
+        objPageAboutScooter.clickOrderButton();
+        objPageAboutScooter.clickOrderConfirm();
+        assertTrue("Ожидамый результат = " + expectedStatusOrder + ".Фактический = " + objPageAboutScooter.getPopUpWindowAfterOrder() + "!",objPageAboutScooter.getPopUpWindowAfterOrder().contains(expectedStatusOrder));
```
**можно лучше**
тебе не обязательно писать "Ожидамый результат" и ".Фактический ". Если проверка не пройдет, она и так выведет какой был ожидаемый, а какой фактический. Так что ты просто сделал работу за нее =)

pavel-ulan reviewed (changes requested):
Неплохо. Есть несколько минорных замечаний которые надо поправить, но в целом работа довольно хорошая =)

senkid committed:
Second-commit.
1. В pom.xml поменял отсылку с 8-ой версии компилятора на 11-ую.
2. Изменил локаторы на более читаемые.
3. Убрал методы с явными ожиданиями, добавил неявное ожидание вначале каждого теста.
4. В CreateOrderTest.java убрал сообщение при проверке ожидаемого и фактического результата.

Senkid12 commented:
Учел замечания в работе во втором коммите

pavel-ulan commented on src/main/java/org/yandex/prakrikum/pageObject/PageAboutScooter.java:
```diff
@@ -0,0 +1,64 @@
+    // кнопка подтверждения заказа
+    private final By orderConfirmation = By.xpath(".//div[@class='Order_Modal__YZ-d3']/div[@class='Order_Buttons__1xGrp']/button[2]");
+//    всплывающее окно после подтверждения заказа
+    private final By popUpWindowAfterOrder = By.xpath("/html/body/div/div/div[2]/div[5]");
```
**можно лучше**
всеравно оставляешь абсолют-пасс локаторы. Все же старайся этого не делать

Senkid12 replied:
Принял, спасибо за ревью кода)

pavel-ulan reviewed (approved):
Да, отличненько 👍
